prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -78,8 +82,11 @@ async def _arun(
 
 
 class CrewAIAgentSubquestions:
-    def __init__(self, model: str) -> None:
+    def __init__(self, model: str, memory: bool = True) -> None:
         self.model = model
+        self.memory = memory
+        if self.memory:
+            self._long_term_memory = LongTermMemory("think-thoroughly-agent")
```

Comment:
Note that, as discussed with @evangriffiths , we won't filter by question (no such clear division on the microchain agent, which also shares memory), but rather filter only on task-datetime.

Reply:
Seems like long_term_memory can be undefined if memory=false, but later in the methods it is accessed unconditionally

Reply:
Great catch! Fixed that